### PR TITLE
LPD-40179 Allow exporting and importing one Page referencing Order Forecasts Chart widget across environments via Headless APIs

### PR DESCRIPTION
**Pull request by michelevigilante**
https://liferay.atlassian.net/browse/LPD-40179

Resent of https://github.com/liferay-commerce/liferay-portal/pull/5665

### CODE REVIEW

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

**Comment by michelevigilante** [comment]
ci:report:91145415

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java [hunk 1]
```diff
@@ -26,24 +34,61 @@ public class CommerceDashboardForecastDisplayContext {
...
+							themeDisplay.getCompanyGroupId());
+			}
+
+			if (assetCategory != null) {
+				assetCategoryIds.add(
+					String.valueOf(assetCategory.getCategoryId()));
+			}
+		}
+
+		return StringUtil.merge(assetCategoryIds);
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java:91]: Burrito

cc @smottal

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java (hunk 1 above)
Comment (smottal) [reply modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java]: I stopped to check immediately for this chain
```
_commerceDashboardForecastPortletInstanceConfiguration.
				assetCategoryExternalReferenceCodes(
				).split(
					StringPool.COMMA
				)
```

Comment (smottal) [reply modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java]: and I think you can do it using TransformUtil

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java [hunk 2]
```diff
@@ -74,12 +82,25 @@ public void processAction(
 		ServiceContext serviceContext = ServiceContextFactory.getInstance(
 			actionRequest);
 
+		List<String> assetCategoryExternalReferenceCodes = new ArrayList<>();
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java:85]: Move it before the `serviceContext` declaration

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java [hunk 3]
```diff
@@ -74,12 +80,24 @@ public void processAction(
...
-			StringUtil.merge(assetCategoryIds));
+			actionRequest, "assetCategoryExternalReferenceCodes",
+			StringUtil.merge(
+				TransformUtil.transform(
+					ArrayUtil.toStringArray(
+						serviceContext.getAssetCategoryIds()),
+					assetCategoryId -> {
+						AssetCategory assetCategory =
+							_assetCategoryLocalService.fetchAssetCategory(
+								GetterUtil.getLong(assetCategoryId));
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java:92]: No need to transform to String and then to Long again. Just use Long values here.

**Comment by michelevigilante** [comment]
ci:test:sf

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java [hunk 4]
```diff
@@ -74,12 +79,23 @@ public void processAction(
...
-			actionRequest, "assetCategoryIds",
-			StringUtil.merge(assetCategoryIds));
+			actionRequest, "assetCategoryExternalReferenceCodes",
+			StringUtil.merge(
+				TransformUtil.transform(
+					serviceContext.getAssetCategoryIds(),
+					assetCategoryId -> {
+						AssetCategory assetCategory =
+							_assetCategoryLocalService.fetchAssetCategory(
+								GetterUtil.getLong(assetCategoryId));
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java:90]: `assetCategoryId` is already a long value.

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:stop

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:forward

**Pull request closed without merging**